### DISCUSSION TITLE
cFS (sometimes) cannot be killed and best practices for shutdown

### DISCUSSION THREAD

**Discussion by bdcmu** [nasa/cFS#552] [Q&A] [answered]
In most circumstances I have found that a running cFS instance can be killed with Ctrl-C or Ctrl-\

However, sometimes I noticed that cFS can become completely un-killable. In these circumstances where cFS will not terminate I have found that nothing works short of a shutdown/restart of the computer. Even liberal use of `kill -9` does nothing to stop cFS (with sudo).

On rare occasions I have noticed that even rebooting the computer causes cFS to refuse to startup on reboot unless a full shutdown of the machine is invoked. A full shutdown and power-on of the computer seems to always work. This may be unrelated to the un-killable issue I care about. 

My questions are:
1. Is this expected behavior?
2. Is there a better way to shutdown cFS cleanly?

NOTE: This is using BootesRC2 on Linux based computers.

**Comment by skliper** [comment, marked as answer]
I seem to recall some changes related to cancellation points (or missing cancellation points) since Bootes-RC2, but from a quick search only found https://github.com/nasa/osal/pull/1028 in the framework.  There may be a missing cancellation point in apps from that time period.  HS had a recent fix: https://github.com/nasa/HS/pull/38.  Caelum also reworked interrupt handling (moved to PSP) to clean up Ctrl-C behavior which may be related.  Either way I haven't observed issues like this in Draco at all (other than the HS issue we fixed).

**Reply by bdcmu** [reply]
Thanks very much!
